src/main/scala/ru/d4team/chat/services/ChatRoomsService.scala

```diff
@@ -33,4 +34,12 @@ final case class ChatRoomsServiceMapImpl(private[services] val rooms: ChatRoomsM
         case None       => TrieMap(channel.id -> channel).some
       }
     }.unit
+
+  def leave(channel: Channel[WebSocketFrame], room: String): UIO[Unit] =
+    ZIO.succeed {
+      rooms.updateWith(room) {
+        case Some(room) => room.subtractOne(channel.id).some.filter(_.nonEmpty)
+        case None       => None
```

Comment:
Maybe log some warning when we try to leave not existed room?

Reply:
Thank you for the good suggestion.
If you don't mind, I would sacrifice logging here for now in favor of concurrency.

I've added a TODO about this to come back and add logging in future.